ground_truth_slam/plugin.cpp

```diff
@@ -26,29 +27,51 @@ class ground_truth_slam : public plugin {
 	}
 
 	void feed_ground_truth(const imu_cam_type *datum) {
-		ullong rounded_time = datum->dataset_time;
+		ullong rounded_time = floor(datum->dataset_time / 10000);
```

Comment:
Is there an implicit cast here?

Reply:
Possibly, but the compiler didn't return any warnings.

Reply:
There is a safe implicit cast here.

While this will work, I prefer to change `ground_truth_slam/data_loading.hpp:46`. This way, you never have to ask what unit the time is in, because it is _always_  in dataset units.

Reply:
I am not familiar with the term - what do you mean @charmoniumQ?

Reply:
By "dataset units" I mean the unit of time our dataset uses (in this case nanoseconds). I think it's bad practice to divide by 1000 just to make up for another place in code we divide by 1000. Instead, I'm suggesting we remove both divisions and use the same units as our input.

Reply:
Good catch. Will fix.